// styled-navbar-component/lib/components/Navbar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Navbar = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styledBaseComponents = require("styled-base-components");

var _styledConfig = require("styled-config");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  display: flex;\n  flex-wrap: wrap;\n  align-items: center;\n  justify-content: space-between;\n  padding: ", ";\n  ", "\n  ", "\n  ", "\n  ", "\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    @media (max-width: ", ") {\n      & > nav:not(:last-child) {\n        display: flex !important;\n        width: 100%;\n        justify-content: space-between;\n      };\n    };\n  "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    background-color: ", " !important;\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    border-color: ", ";\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    color: ", ";\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var color = function color(props) {
  return (0, _styledComponents.css)(_templateObject(), (0, _styledConfig.getColor)(props, 'navbar', 'color'));
};

var borderColor = function borderColor(props) {
  return (0, _styledComponents.css)(_templateObject2(), (0, _styledConfig.getColor)(props, 'navbar', 'borderColor'));
};

var backgroundColor = function backgroundColor(props) {
  return (0, _styledComponents.css)(_templateObject3(), (0, _styledConfig.getColor)(props, 'navbar', 'backgroundColor'));
};
/* istanbul ignore next */


var navbarExpand = function navbarExpand(props) {
  return (props.expandSm || props.expandMd || props.expandLg || props.expandXl) && (0, _styledComponents.css)(_templateObject4(), (0, _styledConfig.getBreakpointSize)(props));
};

var Navbar = (0, _styledComponents.default)(_styledBaseComponents.Nav)(_templateObject5(), function (props) {
  return (0, _styledConfig.getPadding)(props, 'navbar', 'default');
}, function (props) {
  return color(props);
}, function (props) {
  return borderColor(props);
}, function (props) {
  return backgroundColor(props);
}, function (props) {
  return navbarExpand(props);
});
exports.Navbar = Navbar;
Navbar.defaultProps = {
  theme: _styledConfig.theme
};
